dra-evolution/pkg/api/claim_types.go

```diff
@@ -2,78 +2,35 @@ package api
+// device configuration and requirements. It can be referenced in
+// the device requests of a claim to apply these presets.
 // Cluster scoped.
-type ResourceClass struct {
+type DeviceClass struct {
 	metav1.TypeMeta `json:",inline"`
 	// Standard object metadata
 	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	// ControllerName defines the name of the dynamic resource driver that is
-	// used for allocation of a ResourceClaim that uses this class. If empty,
-	// structured parameters are used for allocating claims using this class.
-	//
-	// Resource drivers have a unique name in forward domain order
-	// (acme.example.com).
-	//
-	// This is an alpha field and requires enabling the DRAControlPlaneController
-	// feature gate.
-	//
-	// +optional
-	ControllerName string `json:"controllername,omitempty"`
-
 	// Only nodes matching the selector will be considered by the scheduler
 	// when trying to find a Node that fits a Pod when that Pod uses
 	// a ResourceClaim that has not been allocated yet.
 	//
-	// Setting this field is optional. If null, all nodes are candidates.
-	// +optional
-	SuitableNodes *v1.NodeSelector `json:"suitableNodes,omitempty" protobuf:"bytes,4,opt,name=suitableNodes"`
-
-	// Claim contains options that apply to the entire claim.
-	Claim ResourceClassClaimOptions `json:"claim,omitempty"`
-
-	// Request contains options that apply to requests in a claim.
-	Request ResourceClassRequestOptions `json:"request,omitempty"`
-
-	// DefaultRequests are individual requests for separate resources for a
-	// claim using this class. In contrast to configuration and
-	// requrirements, these requests are only used if the claim does not
-	// specify its own list of requests.
-	//
-	// If empty and the claim doesn't specify any requests either, then
-	// a single, empty request is used.
-	//
-	// +listType=atomic
-	DefaultRequests []ResourceRequest `json:"defaultRequests" protobuf:"bytes,5,name=requests"`
-}
-
-type ResourceClassClaimOptions struct {
-	// Config defines configuration parameters that apply to each claim using this class.
-	// They are ignored while allocating the claim.
+	// Setting this field is optional. If unset, all nodes are candidates.
+	// It is ignored when the claim does not use a control plane controller
+	// for allocation.
 	//
-	// +optional
-	// +listType=atomic
-	Config []ConfigurationParameters `json:"config,omitempty" protobuf:"bytes,3,opt,name=config"`
-
-	// Constraints are additional restrictions for the combination of devices that get
-	// allocated for a claim referencing this class.
+	// This is an alpha field and requires enabling the DRAControlPlaneController
+	// feature gate.
 	//
 	// +optional
-	// +listType=atomic
-	Constraints []Constraint `json:"constraints,omitempty"`
-}
+	SuitableNodes *v1.NodeSelector `json:"suitableNodes,omitempty" protobuf:"bytes,4,opt,name=suitableNodes"`
```

Comment:
This is for classic DRA. It provides some information that comes from the admin which helps the scheduler make better decisions up-front before engaging with the driver in the back-and-forth about potential and unsuitable nodes.

This is not something that users can do in pod or claim because it depends on the cluster.

Reply:
Can you help me understand what the use-case is?

This comes back to "what are classes for?" - if they are how admins configure the cluster and exert "control" then they can't be optional, right?

Reply:
Admins don't use it to override the driver. It's more like providing additional information about a driver to the scheduler. The main use case was:
- Nodes get labeled to indicate which hardware is available.
- The daemonset of a driver uses those labels to avoid running the driver on nodes with no suitable hardware.
- The same node selector is used in the class for that driver.

It's optional because without this hint, the scheduler will still go through the PodSchedulingContext exchange with the driver and thus will find out about suitable nodes. It just might take longer.

Reply:
Can we leave it out for now and bring it back when we have a clear need?

Reply:
Okay.

But just to be clear, we already have this in 1.30. The reason for removal would be to simplify what we need to maintain, not to reduce the implementation work for 1.31.

Reply:
I'm trying to present the SMALLEST POSSIBLE surface so we have the highest confidence that this can move forward.

Reply:
This doesn't need to move forward to beta. It's the "classic DRA" that we want to keep in alpha.